docs/extend/cli_plugins.md

```diff
@@ -0,0 +1,83 @@
+---
```

Comment:
Can we add information here on potentially default paths for detecting plugins?  Had to traverse the code to find it would be good to document probably.

Reply:
Yes, good idea.